### PR TITLE
Show Service events when LoadBalancer cannot be ensured

### LINKED ISSUE #2943: Show Service events when LoadBalancer cannot be ensured
**How to categorize this issue?**

/area ops-productivity
/kind enhancement
/priority normal

**What would you like to be added**:
Currently when the vpn-shoot Service is not ensured, the reconciliation fails with generic error like:

```
Flow "Flow "Shoot cluster reconciliation" encountered task errors: [task "Waiting until the Kubernetes API server can connect to the Shoot workers" failed: retry failed with context deadline exceeded, last error: could not forward to vpn-shoot pod: port forward connection could not be established within five seconds]"]
```

We could add a step like `WaitUntilNginxIngressServiceIsReady` that waits until the vpn-shoot Service `.status.loadBalancer.ingress[]` has elements (when the KonnectivityTunnel feature gate is disabled).
Additionally we could also fetch events associated with the given Service and with the appropriate type and append the event(s) as part of error message.
In this way when the Service cannot be ensured because of quota issue/cloud provider credentials issue, then user would be able to see the reason.

### PR DESCRIPTION
**Pull request by ialidzhikov**
**How to categorize this PR?**

/area ops-productivity
/kind enhancement
/priority normal

**Which issue(s) this PR fixes**:
Fixes #2943

**Release note**:

```noteworthy user
When a Shoot load balancer (vpn-shoot or addons-nginx-ingress-controller) cannot be ensured, gardenlet now fetches the involved object events (with type Warning) and adds them to the error message which is shown in the Shoot status. In this way users and operators will be able to identify better issues in which load balancer cannot be ensured because of invalid cloud provider credentials or another cloud provider issue.
```

### CODE REVIEW

File: pkg/operation/botanist/event.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+func (list SortableEvents) Len() int {
+	return len(list)
+}
+
+func (list SortableEvents) Swap(i, j int) {
+	list[i], list[j] = list[j], list[i]
+}
+
+func (list SortableEvents) Less(i, j int) bool {
```
Comment (rfranzke) [inline pkg/operation/botanist/event.go]: doc strings

File: pkg/operation/botanist/event_test.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+					event3,
+				}
+			)
+
+			sort.Sort(botanist.SortableEvents(events))
+
+			Expect(events).To(Equal([]corev1.Event{event2, event3, event1}))
+		})
+	})
+
```
Comment (rfranzke) [inline pkg/operation/botanist/event_test.go]:
```suggestion
```

**Review by rfranzke — COMMENTED** [review]
Love it, thanks for this great PR!
Only nits
/lgtm

**Review by rfranzke — APPROVED** [review]
/lgtm

**Review by danielfoehrKn — APPROVED** [review]
nice - this will hopefully lower the burden on ops :)

**Pull request merged**